### ISSUE TITLE
hasOne relationship not working, but belongsTo works

### ISSUE THREAD

**Issue by Kleemer** [vuex-orm/vuex-orm#157]
Hi,
I'm trying to use a simple hasOne relationship like this : **element** _hasOne_ **logic**.
However sometimes the **logic** is null. When it's the case, the function HasOne.attach just does not work and it crashes. I then tried to put a **logic** _belongsTo_ **element** and it works correctly.

Am I missing something ?
Thanks in advance for the help !

EDIT : Using belongsTo doesn't work either, the **logic** entities are empty, my bad.
If I replace _hasOne_ by _hasMany_ on both my backend and my entities, it does not crash.

**Comment by kiaking** [comment]
Hi thanks for the report! It sounds like a some edge case defect... Could you paste the model definition of `logic` and `element` and I can try reproduce and fix them? 🔥

**Comment by Kleemer** [comment]
Hi, of course !
Here is the Element Model :
```
class Element extends Model {
  static get entity() {
    return 'elements'
  }

  static fields() {
    return {
      id: this.attr(null),
      title: this.attr(null),
      description: this.attr(''),
      step_id: this.attr(''),
      logic: this.hasOne(Logic, 'element_id')
    }
  }
}
```
And here is the Logic Model :
```
class Logic extends Model {
  static get entity() {
    return 'logics'
  }

  static fields() {
    return {
      id: this.attr(null),
      element_id: this.attr(null),
      logic_triggers: this.hasMany(LogicTrigger, 'logic_id')
    }
  }
}
```
It's not more complicated with that ! Thanks for your quick answers, as always :)

**Comment by kiaking** [comment]
Thanks for the investigation! Sorry I didn't had time to look into this just yet but I'll be soon!

**Comment by Alezco** [comment]
Hey, I just ran into this issue as well. The workaround that @bradstewart posted seems to work! @kiaking Did you have time to work on this? Your help and your time are always appreciated.

**Comment by kiaking** [comment]
@Alezco Yeah I should really! I think I can work on this weekend 👍

**Comment by kiaking** [comment]
This issue is noe resolved at Vuex ORM 0.25.3 🎉 
https://github.com/vuex-orm/vuex-orm/releases/tag/v0.25.3

Thank you all for helping out on this issue! It was wonderful!
